src/out_of_order_engine.cc

```diff
@@ -0,0 +1,414 @@
+#include "out_of_order_engine.h"
+#include "dense_map.h"
+#include "instruction_graph.h"
+#include "system_info.h"
+#include "utils.h"
+
+#include <queue>
+#include <unordered_map>
+#include <variant>
+#include <vector>
+
+#include <matchbox.hh>
+
+namespace celerity::detail::out_of_order_engine_detail {
+
+using target = out_of_order_engine::target;
+using lane_id = out_of_order_engine::lane_id;
+using assignment = out_of_order_engine::assignment;
+
+/// Comparision operator to make priority_queue<instruction*> return concurrent instructions in decreasing priority.
```

Comment:
Typo `Comparision`.

Reply:
My spell checker was okay with that for some reason. Fixed.